cmd/simulator/txs/agent.go

```diff
@@ -53,33 +56,70 @@ func (a issueNAgent[T]) Execute(ctx context.Context) error {
 	}
 
 	txChan := a.sequence.Chan()
+	confirmedCount := 0
+	batchI := 1
+
+	// Tracks the total amount of time waiting for issuing and confirming txs
+	var (
+		totalIssuedTime    time.Duration
+		totalConfirmedTime time.Duration
+	)
+
+	defer func() error {
+		return a.worker.Close(ctx)
+	}()
```

Comment:
why are we returning an error from a deferred function that we're not checking?

Reply:
fixed